### ISSUE TITLE
Add earth engine .map() support for unmixing routines

### ISSUE THREAD

**Issue by earth-chris** [earth-chris/earthlib#12]
Currently, reflectance data need to be composited to an `ee.Image` via an `ee.Reducer` before we can run the unmixing routines (in `ccblc/Unmix.py`). These unmixing routines cannot, at this time, be applied to an image collection using `.map()`. It would be very convenient if we could.

![image](https://user-images.githubusercontent.com/10216182/99868349-a340cb80-2b76-11eb-8e6f-1983b5040b2e.png)

It would be great if we could do this instead:

![image](https://user-images.githubusercontent.com/10216182/99868352-ac319d00-2b76-11eb-8370-e406400579cf.png)

This is not possible right now because there are certain restrictions on which routines run on the client vs which run on the server. Here are some resources that might be useful

- [The EE client-vs-server guide](https://developers.google.com/earth-engine/guides/client_server)
- [Client and server functions](https://developers.google.com/earth-engine/guides/client_server#client-and-server-functions)
- [The deferred execution guide](https://developers.google.com/earth-engine/guides/deferred_execution)

Supporting the use of `.map()` is likely to require re-factoring the unmixing code as it is now written. This will take a lot of research, experimenting, and time. Proceed with caution!
